### PR TITLE
 Upgrade to master of custom-metrics-apiserver

### PR DESCRIPTION
**Pull request by 44past4**

### CODE REVIEW

File: custom-metrics-stackdriver-adapter/adapter.go [hunk 1]
```diff
@@ -18,29 +18,103 @@ package main
...
 
-	// Following deps are not saved in vendor/ by godep save, so import it explicitly
-	_ "cloud.google.com/go"
-	_ "github.com/prometheus/procfs/xfs"
+	adapter "github.com/GoogleCloudPlatform/k8s-stackdriver/custom-metrics-stackdriver-adapter/pkg/adapter/provider"
+	basecmd "github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/cmd"
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
 )
 
+// StackdriverAdapter is an AdapterBase for Stackdriver
```
Comment (kawych) [inline custom-metrics-stackdriver-adapter/adapter.go]: nit: "StackdriverAdapter is an adapter for Stackdriver"

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go [hunk 2]
```diff
@@ -83,14 +86,32 @@ func NewStackdriverProvider(kubeClient *corev1.CoreV1Client, mapper apimeta.REST
...
+func (p *StackdriverProvider) GetMetricBySelector(namespace string, selector labels.Selector, info provider.CustomMetricInfo) (*custom_metrics.MetricValueList, error) {
+	if namespace == "" {
+		return p.GetRootScopedMetricBySelector(info.GroupResource, selector, info.Metric)
+	}
+	return p.GetNamespacedMetricBySelector(info.GroupResource, namespace, selector, info.Metric)
+}
+
 // GetRootScopedMetricByName queries Stackdriver for metrics identified by name and not associated
 // with any namespace. Current implementation doesn't support root scoped metrics.
 func (p *StackdriverProvider) GetRootScopedMetricByName(groupResource schema.GroupResource, name string, escapedMetricName string) (*custom_metrics.MetricValue, error) {
```
Comment (kawych) [inline custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: Did the framework switched from using those to only "GetMetricByName" and "GetMetricBySelector"? Can you limit the visibility of this older functions?

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator_test.go [hunk 3]
```diff
@@ -23,17 +23,17 @@ import (
 	"time"
 
 	"github.com/GoogleCloudPlatform/k8s-stackdriver/custom-metrics-stackdriver-adapter/pkg/config"
-	"github.com/GoogleCloudPlatform/k8s-stackdriver/custom-metrics-stackdriver-adapter/pkg/provider"
-	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
 	sd "google.golang.org/api/monitoring/v3"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (kawych) [inline custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator_test.go]: This doesn't seem to do anything

File: custom-metrics-stackdriver-adapter/adapter.go (hunk 1 above)
Comment (44past4) [reply custom-metrics-stackdriver-adapter/adapter.go]: done

File: custom-metrics-stackdriver-adapter/pkg/config/gce_config.go [hunk 4]
```diff
@@ -17,10 +17,9 @@ limitations under the License.
 package config
 
 import (
-	"fmt"
```
[the remark this answers is not in the document]
Comment (44past4) [reply custom-metrics-stackdriver-adapter/pkg/config/gce_config.go]: done

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go (hunk 2 above)
Comment (44past4) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go]: Yes, it did. Fixed.

File: custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator_test.go (hunk 3 above)
Comment (44past4) [reply custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator_test.go]: Fixed.

**Comment by kawych** [comment]
LGTM

**Pull request merged**